docs/observability/ovn-observability.md

```diff
@@ -0,0 +1,99 @@
+to each packet in the network. Therefore, even though there are ways to see what OVS/OVN is doing with a particular packet, 
+there is no way to know why.
+
+We aim to solve this problem by providing a way for ovn-kubernetes to generate packet samples enriched with metadata 
+that can be easily correlated back to kubernetes objects or other human-readable pieces of information that provide 
+insights of what ovn-kubernetes is doing with a packet and why.
+
+### User-Stories/Use-Cases
+
+- As a user I want to make sure that the network policies/egress firewalls/etc. are correctly enforced in my cluster.
+- As a cluster admin I want to check why some traffic is allowed or dropped.
+
+## How to enable this feature on an OVN-Kubernetes cluster?
+
+To enable this feature, use `--observability` flag with `kind.sh` script or `--enable-observability` flag with `ovnkube` binary.
+
+To see the samples, use `ovnkube-observ` binary, use `-h` to see allowed flags.
+
+This feature requires OVS 3.4 and linux kernel 6.11.
+
+## Workflow Description
+
+- Observability is enabled by setting the `--enable-observability` flag in the `ovnkube` binary.
+- For now all mentioned features are enabled by this flag at the same time.
+- `ovnkube-observ` binary is used to see the samples. Samples are only generated when the real traffic matching the ACLs
+is sent through the OVS. An example output is:
+```
+OVN-K message: Allowed by default allow from local node policy, direction ingress
+src=10.129.2.2, dst=10.129.2.5
+```
+
+## Implementation Details
+
+### User facing API Changes
+
+No API changes were done.
+
+### OVN sampling details
+
+OVN has 3 main db tables that are used for sampling:
+- `Sample_collector`: This table is used to define the sampling collector. It defines the sampling rate and collectorID, 
+which is used to set up collectors in the OVS. 
+- `Sampling_app`: This table is used to set `ID`s for existing OVN sampling applications, that are sent together with the samples.
+- `Sample`: This table is used to define required samples and point to the collectors. 
+Every sample has `Metadata` that is sent together with the sample.
+
+Samples are attached to the other db tables, for now only to ACLs.
+A sample is generated when a packet matches the ACL. Every Sample contains `Sampling_app.ID` and `Sample.Metadata`,
+that is decoded by `go-controller/observability-lib`.
+
+### OVN-Kubernetes Implementation Details
+
+`Sample_collector` and `Sampling_app` are created or cleaned up when the observability is enabled/disabled on startup.
+When one of the supported objects (for example, network policy) is created, ovn-kuberentes generates an nbdb `Sample` for it.
+
+To decode the samples into human-readable information, `go-controller/observability-lib` is used. It finds `Sample`
+by the attached `Sample.Metadata` and then gets corresponding db object based on `Sampling_add.ID` and `Sample.UUID`.
+The message is then constructed using db object `external_ids`.
+
```

Comment:
Maybe we should add a note/known limitation/whatever here saying that sessions established while sampling was enabled for the ACLs they hit will still be sampled after the sampling config is removed (conntrack).  The opposite is also true: if a session was already established before sampling was enabled we won't be getting samples for it afterwards.